popup.js

```diff
@@ -67,18 +69,94 @@ function outputHTMLString(messages) {
  *     scripts.
  * @param {number} i The index of messages to use.
  */
-function fillRemainingHoles(messages, i) {
+function fillRemainingHolesAndMinimizeStyles(messages, i) {
   var html = messages[i].html;
   var frameHoles = messages[i].frameHoles;
   for (var index in frameHoles) {
     if (frameHoles.hasOwnProperty(index)) {
       var frameIndex = frameHoles[index];
       for (var j = 0; j < messages.length; j++) {
         if (messages[j].frameIndex == frameIndex) {
-          fillRemainingHoles(messages, j);
+          fillRemainingHolesAndMinimizeStyles(messages, j);
```

Comment:
It didn't work for the head element because there are style tags that are placed above the head tag (we discussed this a few weeks ago with Chris, and chose to place the style tags that contain fonts and pseudo elements at the top for simplicity).  Therefore when the document is rendered a new head element is created.  I just pushed a fix (writing a test now).

It doesn't work for the html node because setting iframe.contentDocument.documentElement.innerHTML produces a document with a new html element.  Perhaps I should process the html node separately?

It also doesn't work for pseudo elements.  It wouldn't be a simple fix, but definitely doable.  Do you think that's high on the priority list?